In [11]:
import pandas as pd
import numpy as np
import io
import statsmodels.formula.api as smf
import scipy.stats as stats
import re

## Значимость регрессии 

### Задание 1.1

In [13]:
data_string = '''Observations     706         706         706         706    
R2              0.132       0.131       0.128       0.007   
Adjusted.R2     0.121       0.123       0.121       0.002   '''

In [15]:
df = pd.read_csv(io.StringIO(data_string), delimiter='\s+', header=None, index_col=0)
df.reset_index()

0        1        2        3        4
0  Observations  706.000  706.000  706.000  706.000
1            R2    0.132    0.131    0.128    0.007
2   Adjusted.R2    0.121    0.123    0.121    0.002

In [16]:
df = df.transpose()
df

0  Observations     R2  Adjusted.R2
1         706.0  0.132        0.121
2         706.0  0.131        0.123
3         706.0  0.128        0.121
4         706.0  0.007        0.002

In [17]:
# Заполним столбец значением m
df['m'] = [10,8,6,5]
df

0  Observations     R2  Adjusted.R2   m
1         706.0  0.132        0.121  10
2         706.0  0.131        0.123   8
3         706.0  0.128        0.121   6
4         706.0  0.007        0.002   5

In [24]:
df['F_stat'] = (df['R2']/(1-df['R2']))*((df['Observations']-df['m'])/(df['m']-1))
df.round(2)

0  Observations    R2  Adjusted.R2   m  F_nab  F_stat  F_cr
1         706.0  0.13         0.12  10  11.76   11.76  1.89
2         706.0  0.13         0.12   8  15.03   15.03  2.02
3         706.0  0.13         0.12   6  20.55   20.55  2.23
4         706.0  0.01         0.00   5   1.24    1.24  2.38

In [23]:
alpha = 0.05
df['F_cr'] = stats.f.ppf(1-alpha,df['m']-1,df['Observations'] - df['m'])
df.round(2)

0  Observations    R2  Adjusted.R2   m  F_nab  F_stat  F_cr
1         706.0  0.13         0.12  10  11.76   11.76  1.89
2         706.0  0.13         0.12   8  15.03   15.03  2.02
3         706.0  0.13         0.12   6  20.55   20.55  2.23
4         706.0  0.01         0.00   5   1.24    1.24  2.38

In [25]:
df['znachimost'] = df.apply(lambda x : 'znachima' if x['F_nab'] > x['F_cr'] else 'neznachimo', axis = 1)
df

0  Observations     R2  Adjusted.R2   m      F_nab     F_stat      F_cr  \
1         706.0  0.132        0.121  10  11.760369  11.760369  1.893317   
2         706.0  0.131        0.123   8  15.031728  15.031728  2.022681   
3         706.0  0.128        0.121   6  20.550459  20.550459  2.226901   
4         706.0  0.007        0.002   5   1.235398   1.235398  2.384638   

0  znachimost  
1    znachima  
2    znachima  
3    znachima  
4  neznachimo


- значимы все кроме 4 регрессии


### Задание 1.3

In [29]:
data_string = '''Observations            706         706         706         706    
R2                     0.118       0.022       0.015       0.007   
Adjusted.R2            0.110       0.014       0.012       0.002   
Residual.Std.Error   419.371     441.301     441.775     444.018  
F.Statistic          13.387    2.663    5.224      1.314   '''

In [30]:
df = pd.read_csv(io.StringIO(data_string), delimiter='\s+', header=None, index_col=0)
df.reset_index()

0        1        2        3        4
0        Observations  706.000  706.000  706.000  706.000
1                  R2    0.118    0.022    0.015    0.007
2         Adjusted.R2    0.110    0.014    0.012    0.002
3  Residual.Std.Error  419.371  441.301  441.775  444.018
4         F.Statistic   13.387    2.663    5.224    1.314

In [31]:
df = df.transpose()
df

0  Observations     R2  Adjusted.R2  Residual.Std.Error  F.Statistic
1         706.0  0.118        0.110             419.371       13.387
2         706.0  0.022        0.014             441.301        2.663
3         706.0  0.015        0.012             441.775        5.224
4         706.0  0.007        0.002             444.018        1.314

In [34]:
# Заполним столбец значением m
df['m'] = [8,7,3,5]
df

0  Observations     R2  Adjusted.R2  Residual.Std.Error  F.Statistic  m
1         706.0  0.118        0.110             419.371       13.387  8
2         706.0  0.022        0.014             441.301        2.663  7
3         706.0  0.015        0.012             441.775        5.224  3
4         706.0  0.007        0.002             444.018        1.314  5

In [35]:
alpha = 0.01
df['F_cr'] = stats.f.ppf(1-alpha,df['m']-1,df['Observations'] - df['m'])
df.round(2)

0  Observations    R2  Adjusted.R2  Residual.Std.Error  F.Statistic  m  F_cr
1         706.0  0.12         0.11              419.37        13.39  8  2.66
2         706.0  0.02         0.01              441.30         2.66  7  2.83
3         706.0  0.02         0.01              441.78         5.22  3  4.64
4         706.0  0.01         0.00              444.02         1.31  5  3.35

In [38]:
df['znachimost'] = df.apply(lambda x : 'znachima' if x['F.Statistic'] > x['F_cr'] else 'neznachimo', axis = 1)
df

0  Observations     R2  Adjusted.R2  Residual.Std.Error  F.Statistic  m  \
1         706.0  0.118        0.110             419.371       13.387  8   
2         706.0  0.022        0.014             441.301        2.663  7   
3         706.0  0.015        0.012             441.775        5.224  3   
4         706.0  0.007        0.002             444.018        1.314  5   

0      F_cr  znachimost  
1  2.664930    znachima  
2  2.827785  neznachimo  
3  4.635470    znachima  
4  3.346000  neznachimo

- значимы все кроме 2 и 4 регрессий 

### Задание 1.4

In [39]:
data_string = '''Observations            158         158         158        158      158  
R2                     0.992       0.976       0.982      0.053    0.026 
Adjusted.R2            0.992       0.976       0.982      0.035    0.020 
Residual.Std.Error    0.138       0.239       0.209      1.516    1.527 
F.Statistic         3880.407 3185.767 2106.934  2.881  4.158'''

In [40]:
df = pd.read_csv(io.StringIO(data_string), delimiter='\s+', header=None, index_col=0)
df.reset_index()

0         1         2         3        4        5
0        Observations   158.000   158.000   158.000  158.000  158.000
1                  R2     0.992     0.976     0.982    0.053    0.026
2         Adjusted.R2     0.992     0.976     0.982    0.035    0.020
3  Residual.Std.Error     0.138     0.239     0.209    1.516    1.527
4         F.Statistic  3880.407  3185.767  2106.934    2.881    4.158

In [41]:
df = df.transpose()
df

0  Observations     R2  Adjusted.R2  Residual.Std.Error  F.Statistic
1         158.0  0.992        0.992               0.138     3880.407
2         158.0  0.976        0.976               0.239     3185.767
3         158.0  0.982        0.982               0.209     2106.934
4         158.0  0.053        0.035               1.516        2.881
5         158.0  0.026        0.020               1.527        4.158

In [42]:
# Заполним столбец значением m
df['m'] = [6,3,5,4,2]
df

0  Observations     R2  Adjusted.R2  Residual.Std.Error  F.Statistic  m
1         158.0  0.992        0.992               0.138     3880.407  6
2         158.0  0.976        0.976               0.239     3185.767  3
3         158.0  0.982        0.982               0.209     2106.934  5
4         158.0  0.053        0.035               1.516        2.881  4
5         158.0  0.026        0.020               1.527        4.158  2

In [43]:
alpha = 0.01
df['F_cr'] = stats.f.ppf(1-alpha,df['m']-1,df['Observations'] - df['m'])
df.round(2)

0  Observations    R2  Adjusted.R2  Residual.Std.Error  F.Statistic  m  F_cr
1         158.0  0.99         0.99                0.14      3880.41  6  3.14
2         158.0  0.98         0.98                0.24      3185.77  3  4.74
3         158.0  0.98         0.98                0.21      2106.93  5  3.44
4         158.0  0.05         0.04                1.52         2.88  4  3.91
5         158.0  0.03         0.02                1.53         4.16  2  6.80

In [44]:
df['znachimost'] = df.apply(lambda x : 'znachima' if x['F.Statistic'] > x['F_cr'] else 'neznachimo', axis = 1)
df

0  Observations     R2  Adjusted.R2  Residual.Std.Error  F.Statistic  m  \
1         158.0  0.992        0.992               0.138     3880.407  6   
2         158.0  0.976        0.976               0.239     3185.767  3   
3         158.0  0.982        0.982               0.209     2106.934  5   
4         158.0  0.053        0.035               1.516        2.881  4   
5         158.0  0.026        0.020               1.527        4.158  2   

0      F_cr  znachimost  
1  3.139909    znachima  
2  4.744744    znachima  
3  3.444189    znachima  
4  3.911342  neznachimo  
5  6.800161  neznachimo

- значимы все кроме 4 и 5 регрессий 

## Cовместная значимость

### Задача 2.1

In [59]:
data_string = '''Estimate   Std.Error  t.value  Pr>|t|    
(Intercept)  3446.830301   81.839915 42.1167  2e-16 
totwrk         -0.169130    0.018074 -9.3577  2e-16 
age             2.714483    1.472374  1.8436  0.06566   
male           87.108150   35.173210  2.4765  0.01350   
south         102.271833   41.925047  2.4394  0.01496   
smsa          -54.187710   33.193402 -1.6325  0.10303    
yngkid        -13.051272   50.459104 -0.2587  0.79598    
marr          31.360412   42.263190  0.7420  0.45832    
union         11.865664   38.185877  0.3107  0.75610    '''

In [60]:
df=pd.read_csv(io.StringIO(data_string), sep='\s+')
df

Estimate  Std.Error  t.value        Pr>|t|
(Intercept)  3446.830301  81.839915  42.1167  2.000000e-16
totwrk         -0.169130   0.018074  -9.3577  2.000000e-16
age             2.714483   1.472374   1.8436  6.566000e-02
male           87.108150  35.173210   2.4765  1.350000e-02
south         102.271833  41.925047   2.4394  1.496000e-02
smsa          -54.187710  33.193402  -1.6325  1.030300e-01
yngkid        -13.051272  50.459104  -0.2587  7.959800e-01
marr           31.360412  42.263190   0.7420  4.583200e-01
union          11.865664  38.185877   0.3107  7.561000e-01

In [62]:
df['znachimost'] = np.abs(df['Estimate']/df['Std.Error'])
df.round(3)

Estimate  Std.Error  t.value  Pr>|t|  znachimost
(Intercept)  3446.830     81.840   42.117   0.000      42.117
totwrk         -0.169      0.018   -9.358   0.000       9.358
age             2.714      1.472    1.844   0.066       1.844
male           87.108     35.173    2.476   0.014       2.477
south         102.272     41.925    2.439   0.015       2.439
smsa          -54.188     33.193   -1.632   0.103       1.632
yngkid        -13.051     50.459   -0.259   0.796       0.259
marr           31.360     42.263    0.742   0.458       0.742
union          11.866     38.186    0.311   0.756       0.311

In [63]:
alpha = 0.05
t = stats.t.ppf(1-alpha/2, 707-9)
t.round(2)

1.96

In [64]:
df['result'] = df.apply(lambda x: 'znachimo' if x['znachimost'] > t else 'ne znachimo', axis=1 )
df

Estimate  Std.Error  t.value        Pr>|t|  znachimost  \
(Intercept)  3446.830301  81.839915  42.1167  2.000000e-16   42.116739   
totwrk         -0.169130   0.018074  -9.3577  2.000000e-16    9.357641   
age             2.714483   1.472374   1.8436  6.566000e-02    1.843610   
male           87.108150  35.173210   2.4765  1.350000e-02    2.476548   
south         102.271833  41.925047   2.4394  1.496000e-02    2.439397   
smsa          -54.187710  33.193402  -1.6325  1.030300e-01    1.632484   
yngkid        -13.051272  50.459104  -0.2587  7.959800e-01    0.258650   
marr           31.360412  42.263190   0.7420  4.583200e-01    0.742027   
union          11.865664  38.185877   0.3107  7.561000e-01    0.310734   

                  result  
(Intercept)     znachimo  
totwrk          znachimo  
age          ne znachimo  
male            znachimo  
south           znachimo  
smsa         ne znachimo  
yngkid       ne znachimo  
marr         ne znachimo  
union        ne znachimo

- totwrk, male, south являются значимыми переменными 

#### Проверим гипотезы
$$
H_0 = \beta_{smsa} = \beta_{ynkid} = \beta_{marr} = \beta_{union} = 0
$$
$$
H_1 = \beta_{smsa}^2 = \beta_{ynkid}^2 = \beta_{marr}^2 = \beta_{union}^2 > 0 
$$

In [66]:
unrest, rest = np.float_(re.findall(r'([-+]?\d+.\d+)', 'R2   0.131   0.127'))
print(unrest, rest)

0.131 0.127


In [68]:
F_nab = (unrest-rest)/(1-unrest)*(706-9)/4
round(F_nab, 3)

0.802

In [90]:
alpha = 0.05
F_cr = stats.f.ppf(1-alpha, 4, 706-9)
round(F_cr, 3)

2.385

In [89]:
if (F_nab > F_cr ):
    print('Отвергаем H0, коэффициенты совместно значимы')
else:
    print('Не отвергаем H0, коэффициенты совместно незначимы')

Не отвергаем H0, коэффициенты совместно незначимы


## Задание 2.3

In [75]:
data_string = ''' Estimate  Std.Error    
(Intercept) 3587.178292  220.238404 
totwrk        -0.149121    0.026286 
age           -6.258663   11.191863 
I(age^2)       0.106517    0.133767     
male         161.890143   82.177481   
south        112.143584   40.717344  
totwrk:male   -0.034694    0.036279   '''

In [77]:
df = pd.read_csv(io.StringIO(data_string), sep='\s+')
df

Estimate   Std.Error
(Intercept)  3587.178292  220.238404
totwrk         -0.149121    0.026286
age            -6.258663   11.191863
I(age^2)        0.106517    0.133767
male          161.890143   82.177481
south         112.143584   40.717344
totwrk:male    -0.034694    0.036279

In [78]:
df['znachimost'] = np.abs(df['Estimate']/df['Std.Error'])
df.round(3)

Estimate  Std.Error  znachimost
(Intercept)  3587.178    220.238      16.288
totwrk         -0.149      0.026       5.673
age            -6.259     11.192       0.559
I(age^2)        0.107      0.134       0.796
male          161.890     82.177       1.970
south         112.144     40.717       2.754
totwrk:male    -0.035      0.036       0.956

In [83]:
alpha = 0.01
t = stats.t.ppf(1-alpha/2, 706-7)
t.round(2)

2.58

In [84]:
df['result'] = df.apply(lambda x: 'znachimo' if x['znachimost'] > t else 'ne znachimo', axis=1 )
df

Estimate   Std.Error  znachimost       result
(Intercept)  3587.178292  220.238404   16.287706     znachimo
totwrk         -0.149121    0.026286    5.673020     znachimo
age            -6.258663   11.191863    0.559215  ne znachimo
I(age^2)        0.106517    0.133767    0.796288  ne znachimo
male          161.890143   82.177481    1.970006  ne znachimo
south         112.143584   40.717344    2.754197     znachimo
totwrk:male    -0.034694    0.036279    0.956311  ne znachimo

- totwrk, south являются значимыми переменными 

#### Проверим гипотезы
$$
H_0 = \beta_{male} = \beta_{male*totwrk} = 0
$$
$$
H_1 = \beta_{male}^2 = \beta_{male*totwrk}^2 > 0
$$

In [81]:
unrest, rest = np.float_(re.findall(r'([-+]?\d+.\d+)', 'R2  0.129   0.119'))
print(unrest, rest)

0.129 0.119


In [86]:
F_nab = (unrest-rest)/(1-unrest)*(706-7)/5
round(F_nab, 3)

1.605

In [87]:
alpha = 0.01
F_cr = stats.f.ppf(1-alpha, 4, 706-7)
round(F_cr, 3)

3.346

- не отвергаем гипотезу $H_0$, коэффициенты совместно незначимы 

### Задание 3.1

In [163]:
pd.read_csv('https://raw.githubusercontent.com/artamonoff/Econometrica/master/python-notebooks/data-csv/Electricity.csv')

cost        q        pl      sl      pk      sk      pf      sf
0     0.2130      8.0   6869.47  0.3291  64.945  0.4197  18.000  0.2512
1     3.0427    869.0   8372.96  0.1030  68.227  0.2913  21.067  0.6057
2     9.4059   1412.0   7960.90  0.0891  40.692  0.1567  41.530  0.7542
3     0.7606     65.0   8971.89  0.2802  41.243  0.1282  28.539  0.5916
4     2.2587    295.0   8218.40  0.1772  71.940  0.1623  39.200  0.6606
..       ...      ...       ...     ...     ...     ...     ...     ...
153   6.8293    946.6  10642.16  0.0883  43.600  0.1914  51.463  0.7203
154   3.7605    377.0   7432.24  0.2117  74.120  0.2274  33.436  0.5609
155   3.9822    391.0   5826.04  0.1926  78.288  0.0924  44.633  0.7151
156  30.1880   5317.0   9586.63  0.0845  78.008  0.2009  41.840  0.7147
157  67.8562  15220.0   6986.09  0.1451  74.025  0.2596  25.662  0.5945

[158 rows x 8 columns]

In [165]:
fitted = smf.ols(data=df, formula='np.log(cost)~np.log(q)+I(np.log((q)**2))+np.log(pl)+np.log(pk)+np.log(pf)').fit()
fitted.params

PatsyError: Error evaluating factor: NameError: name 'pl' is not defined
    np.log(cost)~np.log(q)+I(np.log((q)**2))+np.log(pl)+np.log(pk)+np.log(pf)
                                             ^^^^^^^^^^

<class 'statsmodels.iolib.table.SimpleTable'>

In [119]:
data_string = '''coef	std.err	t	P>|t|	[0.005	0.995]
Intercept	-6.7387	0.706	-9.541	0.000	-8.581	-4.896
np.log(q)	0.4030	0.032	12.734	0.000	0.320	0.486
I(np.log(q)**2)	0.0304	0.002	14.024	0.000	0.025	0.036
np.log(pl)	0.1461	0.070	2.073	0.040	-0.038	0.330
np.log(pk)	0.1571	0.058	2.721	0.007	0.007	0.308
np.log(pf)	0.6847	0.043	16.043	0.000	0.573	0.796
'''

In [120]:
df = pd.read_csv(io.StringIO(data_string), sep='\s+')
df

coef  std.err       t  P>|t|  [0.005  0.995]
Intercept       -6.7387    0.706  -9.541  0.000  -8.581  -4.896
np.log(q)        0.4030    0.032  12.734  0.000   0.320   0.486
I(np.log(q)**2)  0.0304    0.002  14.024  0.000   0.025   0.036
np.log(pl)       0.1461    0.070   2.073  0.040  -0.038   0.330
np.log(pk)       0.1571    0.058   2.721  0.007   0.007   0.308
np.log(pf)       0.6847    0.043  16.043  0.000   0.573   0.796

In [121]:
df['result']=df.apply(lambda x: 'значимо' if abs(x['t'])>t_cr else 'не значимо', axis =1)
df

coef  std.err       t  P>|t|  [0.005  0.995]      result
Intercept       -6.7387    0.706  -9.541  0.000  -8.581  -4.896     значимо
np.log(q)        0.4030    0.032  12.734  0.000   0.320   0.486     значимо
I(np.log(q)**2)  0.0304    0.002  14.024  0.000   0.025   0.036     значимо
np.log(pl)       0.1461    0.070   2.073  0.040  -0.038   0.330  не значимо
np.log(pk)       0.1571    0.058   2.721  0.007   0.007   0.308     значимо
np.log(pf)       0.6847    0.043  16.043  0.000   0.573   0.796     значимо

- коэффициенты log(q), log^2(q), log(pk), log (pf) являются значимыми 

###  Задание 3.1.1

#### Проверим гипотезы
$$
H_0 = \beta_{pf} = \beta_{pl} = \beta_{pk} = 1
$$
$$
H_1 = \beta_{pf} = \beta_{pl} = \beta_{pk} > 1
$$

In [161]:
pd.read_csv('https://raw.githubusercontent.com/artamonoff/Econometrica/master/python-notebooks/data-csv/Electricity.csv')

cost        q        pl      sl      pk      sk      pf      sf
0     0.2130      8.0   6869.47  0.3291  64.945  0.4197  18.000  0.2512
1     3.0427    869.0   8372.96  0.1030  68.227  0.2913  21.067  0.6057
2     9.4059   1412.0   7960.90  0.0891  40.692  0.1567  41.530  0.7542
3     0.7606     65.0   8971.89  0.2802  41.243  0.1282  28.539  0.5916
4     2.2587    295.0   8218.40  0.1772  71.940  0.1623  39.200  0.6606
..       ...      ...       ...     ...     ...     ...     ...     ...
153   6.8293    946.6  10642.16  0.0883  43.600  0.1914  51.463  0.7203
154   3.7605    377.0   7432.24  0.2117  74.120  0.2274  33.436  0.5609
155   3.9822    391.0   5826.04  0.1926  78.288  0.0924  44.633  0.7151
156  30.1880   5317.0   9586.63  0.0845  78.008  0.2009  41.840  0.7147
157  67.8562  15220.0   6986.09  0.1451  74.025  0.2596  25.662  0.5945

[158 rows x 8 columns]

PatsyError: Error evaluating factor: NameError: name 'cost' is not defined
    np.log(cost)~np.log(q)+I(np.log(q)**2)+np.log(pl)+np.log(pk)+np.log(pf)
    ^^^^^^^^^^^^

### Задача 3.2

In [148]:
df = pd.read_csv('https://raw.githubusercontent.com/artamonoff/Econometrica/master/python-notebooks/data-csv/Labour.csv')

In [149]:
fitted = smf.ols(data=df, formula='np.log(output)~np.log(capital)+np.log(labour)+np.log(wage)').fit()
fitted.params

Intercept         -5.007316
np.log(capital)    0.149343
np.log(labour)     0.720410
np.log(wage)       0.921390
dtype: float64

In [151]:
fitted.summary(alpha=0.05).tables[1]

<class 'statsmodels.iolib.table.SimpleTable'>

In [152]:
data_string = '''coef	std.err	t	P>|t|	[0.025	0.975]
Intercept	-5.0073	0.221	-22.649	0.000	-5.442	-4.573
np.log(capital)	0.1493	0.015	10.141	0.000	0.120	0.178
np.log(labour)	0.7204	0.019	37.487	0.000	0.683	0.758
np.log(wage)	0.9214	0.058	16.001	0.000	0.808	1.034
'''

In [153]:
df = pd.read_csv(io.StringIO(data_string), sep='\s+')
df

coef  std.err       t  P>|t|  [0.025  0.975]
Intercept       -5.0073    0.221 -22.649    0.0  -5.442  -4.573
np.log(capital)  0.1493    0.015  10.141    0.0   0.120   0.178
np.log(labour)   0.7204    0.019  37.487    0.0   0.683   0.758
np.log(wage)     0.9214    0.058  16.001    0.0   0.808   1.034

In [154]:
df['result']=df.apply(lambda x: 'значимо' if abs(x['t'])>t_cr else 'не значимо', axis =1)
df

coef  std.err       t  P>|t|  [0.025  0.975]   result
Intercept       -5.0073    0.221 -22.649    0.0  -5.442  -4.573  значимо
np.log(capital)  0.1493    0.015  10.141    0.0   0.120   0.178  значимо
np.log(labour)   0.7204    0.019  37.487    0.0   0.683   0.758  значимо
np.log(wage)     0.9214    0.058  16.001    0.0   0.808   1.034  значимо

- log(capital), log(labour), log(wage) являются значимыми коэффициентами 

### Задача 3.2.1

#### Проверим гипотезы
$$
H_0 : \beta_{capital} = \beta_{labour} = \beta_{wage} = 1
$$
$$
H_1 : \beta_{capital}^2 = \beta_{labour}^2 = \beta_{wage}^2 ≠ 1
$$

In [159]:
df = pd.read_csv('https://raw.githubusercontent.com/artamonoff/Econometrica/master/python-notebooks/data-csv/Labour.csv')


In [169]:
fitted_rest = smf.ols(data=df, formula='np.log(output)~np.log(capital)+np.log(labour)+(1-np.log(capital)+np.log(labour))').fit()
fitted_rest.params

Intercept         -1.711459
np.log(capital)    0.207570
np.log(labour)     0.714847
dtype: float64

In [168]:
fitted_unrest = smf.ols(data=df, formula='np.log(output)~np.log(capital)+np.log(labour)+np.log(wage)').fit()
fitted_unrest.params

Intercept         -5.007316
np.log(capital)    0.149343
np.log(labour)     0.720410
np.log(wage)       0.921390
dtype: float64

In [175]:
R2_rest = fitted_rest.rsquared  
R2_rest.round(3)

0.838

In [173]:
R2_unrest = fitted_unrest.rsquared   
R2_unrest.round(3)

0.888

In [182]:
n = 569
m = 4 
F_nab = (R2_unrest - R2_rest)/(1-R2_unrest)*(569-m)/1
F_nab.round(3)

256.021

In [183]:
F_cr = stats.f.ppf(1-0.05, 1, 569-4)
F_cr.round(3)

3.858

In [184]:
if (F_nab > F_cr ):
    print('Отвергаем H0, коэффициенты совместно значимы')
else:
    print('Не отвергаем H0, коэффициенты совместно незначимы')

Отвергаем H0, коэффициенты совместно значимы


- F_cr = 3.858
- $H_0$

### Задача 3.2.2

#### Проверим гипотезы
$$
H_0 : \beta_{labour} = \beta_{wage} = 0
$$
$$
H_1 : \beta_{labour}^2 = \beta_{wage}^2 ≠ 0
$$

In [187]:
fitted = smf.ols(data=df, formula='np.log(output)~np.log(capital)+np.log(labour)+np.log(wage)').fit()
fitted.params

Intercept         -5.007316
np.log(capital)    0.149343
np.log(labour)     0.720410
np.log(wage)       0.921390
dtype: float64

In [197]:
F_nab = fitted.f_test( 'np.log(labour)=np.log(wage)' ) 
round(F_nab.fvalue, 3)

11.081

In [201]:
F_cr = stats.f.ppf(1-0.05,2,569-4)
F_cr.round(3)

3.012

- F_cr = 3.012
- $H_0$ отвергаем

### Задача 3.2.3

#### Проверим гипотезы
$$
H_0 : \beta_{capital} = \beta_{labour} = \beta_{wage} = 0
$$
$$
H_1 : \beta_{capital}^2 = \beta_{labour}^2 = \beta_{wage}^2 ≠ 0
$$

In [207]:
F_nab = fitted.f_test('np.log(capital)=np.log(labour)=np.log(wage)')
round(F_nab.fvalue, 3)

192.367

In [208]:
F_cr = stats.f.ppf(1-0.05,2,569-4)
F_cr.round(3)

3.012

- $H_0$ снова отвергается